automationexcersise/conftest.py

```diff
@@ -0,0 +1,142 @@
+import pytest
+import json
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+import time
+
+@pytest.fixture()
+def get_driver(get_settings):
+    driver = webdriver.Firefox()
+    driver.implicitly_wait(get_settings["wait_duration"])
+    return driver
+    # yield driver
+    # driver.quit()
+
+
+@pytest.fixture()
+def get_settings():
+    with open("config.json") as f:
+        data = json.load(f)
+    return data["settings"]
+
+@pytest.fixture()
+def get_xpaths():
+    with open("config.json") as f:
+        data = json.load(f)
+    return data["xpaths"]
+
+
+@pytest.fixture()
+def get_signup(get_xpaths, get_driver, get_settings):
+    driver = get_driver
+    driver.get(get_settings["base_url"])
+
+    assert WebDriverWait(driver, 5).until(EC.presence_of_element_located((By.XPATH, get_xpaths["home"]["home_button"])))
+    login_button = driver.find_element(By.XPATH, get_xpaths["login"]["login_subpage"])
+    login_button.click()
+
+    assert WebDriverWait(driver, 5).until(EC.presence_of_element_located((By.XPATH, "//div[@class='signup-form']")))
+    signup_name = driver.find_element(By.XPATH, "//input[@data-qa='signup-name']")
+    signup_email = driver.find_element(By.XPATH, "//input[@data-qa='signup-email']")
+    signup_button = driver.find_element(By.XPATH, "//button[@data-qa='signup-button']")
```

Comment:
Remember what I told you about "stale" elements. It's best to get the element right before interacting with it.

Reply:
Fixed